### ISSUE TITLE
[Bug] Site toggle won't save after reload

### ISSUE THREAD

**Issue by AKA-Beastmaster** [darkreader/darkreader#6966]
# Bug Report

## Bug Description

Whenever I disable or enable a site in dark reader, it doesn't stay enabled/disabled after reloading the tab
## Website Address

This seems to happen on all websites
## Steps To Reproduce

Open a website
Open darkreader
Toggle the site off
Reload the tab
Notice how the site has been re-enabled 

## Expected Behavior

The site should stay toggled off

## Actual Behavior

The site stays on

## Screenshots

(not screenshot, felt it would be better to show with a video)
https://user-images.githubusercontent.com/70001656/135553820-8266de20-9d02-479c-bcbf-58c832937d4a.mp4

## System Information

- OS:  Pop!_OS 21.04
- Browser:  Firefox 91.0 (64-bit)
- Dark Reader Version:  version 5 preview (4.9.36)

## Additional Context

This may be unrelated, but I have been using darkreader for a while, so my sitelist is pretty long (not sure how long though)

**Comment by cipitaua** [comment]
I confirm the bug, and I also have a long "sitelist". Moreover if I press "off" button and then "on" button, it stays on off state, and I have to restart firefox to get darkreader enabled.

**Comment by refola** [comment]
I also confirm the bug on my system. More precisely, all changes to the site list are reverted upon changing, opening, or reloading tabs. It's like the extension suddenly developed anterograde amnesia. It works in a fresh-ish Firefox profile, so I suspect it has to do with old data somehow interfering.

**System information**

- OS: Manjaro Linux (rolling, Arch-based)
- Browser: Firefox 92.0.1 (64-bit)
- Dark Reader Version: 4.9.37.1

**Debug output**

Here's what I get from stumbling around with Firefox's "Debug Add-ons" console when removing a site from the end of the sites list (click Dark Reader button, click "Site list", `shift+tab` to select last, `delete` to clear, `tab` to leave that cell and attempt committing the removal, click icon to close pop-up):

```
Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. 3 index.js:2485
    sendMessage moz-extension://e31879ac-1415-4d69-a826-f62034ece4ac/background/index.js:2485
Promise rejected after context unloaded: Actor 'Conduits' destroyed before query 'RuntimeMessage' was resolved
index.js:1312
    disconnect moz-extension://e31879ac-1415-4d69-a826-f62034ece4ac/ui/popup/index.js:1312
```

The count in the first line, "3", is always 3x however many sites I tried to remove from the list if I keep repeating `shift+tab` and `delete`.

If I click the console instead of using `tab` and clicking the Dark Reader icon to close the pop-up, I get this slightly more verbose output:

```
Promise rejected after context unloaded: Actor 'Conduits' destroyed before query 'RuntimeMessage' was resolved
index.js:1312
    disconnect moz-extension://e31879ac-1415-4d69-a826-f62034ece4ac/ui/popup/index.js:1312
Promise rejected after context unloaded: Actor 'Conduits' destroyed before query 'RuntimeMessage' was resolved
index.js:1267
    changeSettings moz-extension://e31879ac-1415-4d69-a826-f62034ece4ac/ui/popup/index.js:1267
Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. 3 index.js:2485
    sendMessage moz-extension://e31879ac-1415-4d69-a826-f62034ece4ac/background/index.js:2485
```

If I try to enable time-based automation (6:00PM to 9:00AM, presumably default since I've never used automatic mode switching), I get the same lack of efficacy as for changing the site list, but with different console output:

```
Error: Warning processing colors.accentcolor: Unsupported colors property, use 'theme.colors.frame', this alias is ignored in Firefox >= 70. index.js:3335
    setWindowTheme moz-extension://e31879ac-1415-4d69-a826-f62034ece4ac/background/index.js:3335
Error: Warning processing colors.textcolor: Unsupported color property, use 'theme.colors.tab_background_text', this alias is ignored in Firefox >= 70. index.js:3335
    setWindowTheme moz-extension://e31879ac-1415-4d69-a826-f62034ece4ac/background/index.js:3335
Unchecked lastError value: Error: Could not establish connection. Receiving end does not exist. 3 index.js:2485
    sendMessage moz-extension://e31879ac-1415-4d69-a826-f62034ece4ac/background/index.js:2485
Promise rejected after context unloaded: Actor 'Conduits' destroyed before query 'RuntimeMessage' was resolved
index.js:1312
    disconnect moz-extension://e31879ac-1415-4d69-a826-f62034ece4ac/ui/popup/index.js:1312
```

On the (working) fresh install of Dark Reader in a fresh-ish Firefox profile, changing either the sites list or automation gives this error in the console, hence the same can *probably* be ignored for this bug:

```
Promise rejected after context unloaded: Actor 'Conduits' destroyed before query 'RuntimeMessage' was resolved
index.js:1312
    disconnect moz-extension://9b10ccab-f5dd-401b-bc1f-6d76f0ef6114/ui/popup/index.js:1312
```

Source code links thanks to `grep --recursive` because the Firefox add-on's `index.js` doesn't reflect the file structure of this repo:

- `index.js:1267` is [src/ui/connect/connector.ts:73](https://github.com/darkreader/darkreader/blob/master/src/ui/connect/connector.ts#L73)
- `index.js:1312` is [src/ui/connect/connector.ts:129](https://github.com/darkreader/darkreader/blob/master/src/ui/connect/connector.ts#L129)
- `index.js:2485` is [src/background/tab-manager.ts:274](https://github.com/darkreader/darkreader/blob/master/src/background/tab-manager.ts#L274)
- `index.js:3335` is [src/background/window-theme.ts:85](https://github.com/darkreader/darkreader/blob/master/src/background/window-theme.ts#L85)

I have no idea how useful this output is. Please tell me if there's something I can do (e.g., trying a beta version, opening a more verbose log, more sophisticated use of Firefox's extension debugger, etc) to provide more useful data.

**Additional bug effect**

@cipitaua, I have the same issue of being unable to re-enable Dark Reader. A workaround is to disable and re-enable the extension.

**Comment by DUOLabs333** [comment]
I have the same problem -- I thought I did something wrong and it was only affecting me.

**Comment by cipitaua** [comment]
this is a very serious bug, how still "No one assigned" is possible?

**Comment by jehrenzweig** [comment]
Yep, I just noticed this was happening to me in Firefox (92.0.1, 64-bit) on MacOS Catalina.  The only workaround is to disable the Dark Reader extension... I know that seems drastic, but I'm dealing with a benefits open enrollment form & the checkboxes aren't visible in the Dark Reader "_Dark_" style.

**Comment by Gusted** [comment]
Hi everyone!

This seems to be actually more related to just #6873 whereby a new version(4.9.39) is currently being pushed to the web stores.
Let's wait for that and check if the issue has be solved trough that.


@cipitaua I'm going to hide your comment and pretend I haven't read it. I hope you can understand why, otherwise please send a email to `williamzijl7@hotmail.com` as such discussion is off-topic.

Regards,
Gusted

**Comment by AKA-Beastmaster** [comment]
> Hi everyone!
> 
> This seems to be actually more related to just #6873 whereby a new version(4.9.39) is currently being pushed to the web stores. Let's wait for that and check if the issue has be solved trough that.
> 
> @cipitaua I'm going to hide your comment and pretend I haven't read it. I hope you can understand why, otherwise please send a email to `williamzijl7@hotmail.com` as such discussion is off-topic.
> 
> Regards, Gusted

Alright, 
Thank you.

**Comment by Zorrototo** [comment]
In the meantime, disabling and enabling the extension (v4.9.37.1) again in the browser (Firefox) fixes the issue for me so the changes I made for a website are persistent. Hope next version fixes it.

**Comment by OmarNamis** [comment]
I just received the update, and the issue is fixed for me.

**Comment by DUOLabs333** [comment]
I didn't get it yet -- did it get pushed out for Firefox (unrelated, but I wish Firefox made it easier to install self-compiled extensions, instead of having to use special Linux directories that can't override already installed versions).

**Comment by OmarNamis** [comment]
@DUOLabs333 Try installing it from here:
https://addons.mozilla.org/en-US/addon/darkreader/versions/

**Comment by Zorrototo** [comment]
So far it seems to be fixed with version 4.9.39

**Comment by Gusted** [comment]
4.9.39 got pushed for firefox(finally) and Chrome was already updated 2 days ago. Closing.
